### PR TITLE
[Caffe2] [ThreadPool] WorkersPool uses atomic writes to task_

### PR DESCRIPTION
**Pull request by ajtulloch**
cc @Maratyszcza

### CODE REVIEW

File: caffe2/utils/threadpool/WorkersPool.h [hunk 1]
```diff
@@ -277,7 +277,7 @@ class alignas(kGEMMLOWPCacheLineSize) Worker {
       case State::HasWork:
         // Got work to do! So do it, and then revert to 'Ready' state.
         DCHECK(task_);
-        task_->Run();
+        (*task_).Run();
```
Comment (Maratyszcza) [inline caffe2/utils/threadpool/WorkersPool.h]: This change seems unrelated

Comment (Maratyszcza) [reply caffe2/utils/threadpool/WorkersPool.h]: Nvm, its due to `atomic` type

**Review by Maratyszcza — APPROVED** [review]
LGTM, but please wait for all builds to pass before merging

**Comment by ajtulloch** [comment]
@caffe2bot test this please.
@pytorch/buildbot test this please.
@onnxbot test this please.

**Comment by ajtulloch** [comment]
@caffe2bot test this please.
@pytorch/buildbot test this please.
@onnxbot test this please.
